### PR TITLE
Made Energy Generation for Kuki and Yelan more accurate

### PR DESCRIPTION
**Pull request by honeyed-lemons**
## Description
 - Updated energy generation to be more accurate to calculations from Ubatcha
## Type of changes

- [ ] Bug fix
- [ ] New feature 
- [x] Enhancement
- [ ] Documentation

## Checklist:

- [x]  My code follows the style guidelines of this project
- [x] My pull request is unique and no other pull requests have been opened for these changes
- [x] I have read the [Contributing note](https://github.com/Grasscutters/Grasscutter/blob/stable/CONTRIBUTING.md) and [Code of conduct](https://github.com/Grasscutters/Grasscutter/blob/development/CODE_OF_CONDUCT.md)
- [x] I am responsible for any copyright issues with my code if it occurs in the future.

### CODE REVIEW

File: src/main/resources/defaults/data/SkillParticleGeneration.json [hunk 1]
```diff
@@ -555,7 +559,7 @@
 		"amountList": [
 			{
 				"value": 1,
-				"chance": 100
+				"chance": 45
```
Comment (GanyusLeftHorn) [inline src/main/resources/defaults/data/SkillParticleGeneration.json:562]: The zero particle case is handled by the client. Shinobu always generates exactly one particle when the client invokes GENERATE_ELEM_BALL.

File: src/main/resources/defaults/data/SkillParticleGeneration.json [hunk 2]
```diff
@@ -515,7 +515,11 @@
 		"amountList": [
 			{
 				"value": 4,
-				"chance": 100
+				"chance": 70
+			},
+			{
+				"value": 3,
+				"chance": 30
```
Comment (GanyusLeftHorn) [inline src/main/resources/defaults/data/SkillParticleGeneration.json:522]: Do you have a source for this? I know these are the rates that some leaks mentioned, but `BinOutput/Ability/Temp/AvatarAbilities/ConfigAbility_Avatar_Yelan.json` does not support this. The data in there says that she always generates 12 energy, i.e. 4 hydro particles.

File: src/main/resources/defaults/data/SkillParticleGeneration.json (hunk 1 above)
Comment (honeyed-lemons) [reply src/main/resources/defaults/data/SkillParticleGeneration.json:562]: Ah, ok, got it

File: src/main/resources/defaults/data/SkillParticleGeneration.json (hunk 2 above)
Comment (honeyed-lemons) [reply src/main/resources/defaults/data/SkillParticleGeneration.json:522]: Theres always a chance it changed from beta, however ubacha has claimed that theres a chance for it to generate 3-4 depending on hold and press skill. Heres the tweet. Unless his source lied it'd be weird if it generated anything else https://twitter.com/Ubatcha1/status/1509954923357777921?s=20&t=x5TRifdlh_japR82Ju4Oyg

Comment (GanyusLeftHorn) [reply src/main/resources/defaults/data/SkillParticleGeneration.json:522]: Yes, that is the leak I mentioned. There was also an earlier one about her always generating four, from an earlier version of the beta. It is possible that they just went back to that for the release.

For now, I would suggest sticking to what the data in `BinOutput` says. If tests on the live server show different behavior, we can always change it then.

Comment (honeyed-lemons) [reply src/main/resources/defaults/data/SkillParticleGeneration.json:522]: 👍 got it

**Pull request closed without merging**